qiskit/aqua/operators/state_functions/dict_state_fn.py

```diff
@@ -0,0 +1,303 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" An Object to represent State Functions constructed from Operators """
+
+from typing import Union
+import itertools
+import numpy as np
+from scipy import sparse
+
+from qiskit.result import Result
+from qiskit.circuit import ParameterExpression
+
+from ..operator_base import OperatorBase
+from . import StateFn
+from ..combo_operators import ListOp
+
+
+class DictStateFn(StateFn):
```

Comment:
Because these are not counts - they are just a sparse representation of a statefn which is implemented by a hashtable. The values here can be the amplitudes of a statevector, or the counts, or anything else.

Reply:
I know I need to beef up the docs to explain this